lib/platform/client/proxy_http.js

```diff
@@ -1,164 +1,228 @@
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+(function() {
+    var Http    = require('../../http');
+    var utils   = require('../../utils');
+    var SDK_VERSION = require('../../../package.json').version;
+
+    var root = exports || this;
+    
+    var getHeaders = function(headersString) {
+        var headers = {};
+        var headerLines = headersString.split("\n");
+        for(var i = 0; i < headerLines.length; i++) {
+            if (utils.trim(headerLines[i]) !== "") {
+                var headerParts = headerLines[i].split(": ");
+                headers[headerParts[0]] = headerParts[1];
+            }
+        }
+
+        return headers;
+    };
+    
+    // parseUri 1.2.2
+    // (c) Steven Levithan <stevenlevithan.com>
+    // MIT License
+    function parseUri (str) {
+        var o   = parseUri.options,
+            m   = o.parser[o.strictMode ? "strict" : "loose"].exec(str),
+            uri = {},
+            i   = 14;
+
+        while (i--) {
+            uri[o.key[i]] = m[i] || "";
+        }
+
+        uri[o.q.name] = {};
+        uri[o.key[12]].replace(o.q.parser, function ($0, $1, $2) {
+            if ($1) {
+                uri[o.q.name][$1] = $2;
+            }
+        });
+
+        return uri;
+    }
+
+    parseUri.options = {
+        strictMode: false,
+        key: ["source","protocol","authority","userInfo","user","password","host","port","relative","path","directory","file","query","anchor"],
+        q:   {
+            name:   "queryKey",
+            parser: /(?:^|&)([^&=]*)=?([^&]*)/g
+        },
+        parser: {
+            strict: /^(?:([^:\/?#]+):)?(?:\/\/((?:(([^:@]*)(?::([^:@]*))?)?@)?([^:\/?#]*)(?::(\d*))?))?((((?:[^?#\/]*\/)*)([^?#]*))(?:\?([^#]*))?(?:#(.*))?)/,
+            loose:  /^(?:(?![^:@]+:[^:@\/]*@)([^:\/?#.]+):)?(?:\/\/)?((?:(([^:@]*)(?::([^:@]*))?)?@)?([^:\/?#]*)(?::(\d*))?)(((\/(?:[^?#](?![^?#\/]*\.[^?#\/.]+(?:[?#]|$)))*\/?)?([^?#\/]*))(?:\?([^#]*))?(?:#(.*))?)/
+        }
+    };
+    
+    var specials = /[.*+?|()\[\]{}\\$\^]/g; // .*+?|()[]{}\$^
+    var escape = function(str) {
+        str = str || "";
+        return str.replace(specials, "\\$&");
+    };
+
+    root.ProxyHttp = Http.extend({
+        init: function(prefix) {
+            this.prefix = prefix;
+            this._super();
+        },
+
+        makeRequest: function(url, message) {
+            // Add our original destination to to headers,
+            // as some proxy implementations would rather
+            // use this.
+            message.headers["X-ProxyDestination"] = url;
+            message.headers["Splunk-Client"] = "splunk-sdk-javascript/" + SDK_VERSION;
+            if(message.response_timeout != undefined){
+                message.headers["Response-Timeout"] = message.response_timeout;

```

Comment:
same comment as above, not sure this header makes sense to me

Reply:
From proxy_http.js we are passing data to the cli.js through which we communicate with apis using needle. we have added this header to make client side code work similar as backend where we replaced abort method with response_timeout parameter(node_http.js file).

Reply:
Interesting, I think I need to dig in more as to how the cli and this client interact

Reply:
as per our discussion we have removed the header and now we'll consider the max of `timeout` and `response_timeout` for aborting request.